geoalchemy2/admin/dialects/sqlite.py

```diff
@@ -28,19 +44,73 @@ def load_spatialite(dbapi_conn, connection_record):
     dbapi_conn.enable_load_extension(True)
     dbapi_conn.load_extension(os.environ["SPATIALITE_LIBRARY_PATH"])
     dbapi_conn.enable_load_extension(False)
-    dbapi_conn.execute("SELECT InitSpatialMetaData();")
+    try:
+        databases = [
+            i[-1].endswith(".gpkg") for i in dbapi_conn.execute("PRAGMA database_list;").fetchall()
+        ]
+        is_GeoPkg = any(databases)
+    except Exception:
+        is_GeoPkg = False
+    if is_GeoPkg:
+        if not dbapi_conn.execute("SELECT CheckGeoPackageMetaData();").fetchone()[0]:
+            # This only works on the main database
+            dbapi_conn.execute("SELECT gpkgCreateBaseTables();")
```

Comment:
As `InitSpatialMetadata()` populates the srs table, I think it is good to call `gpkgInsertEpsgSRID()` for the geopackage at this, so that behaviour is the same.

Users that don't want to have the initialization could just roll their own `load_spatialite` handler

Reply:
I agree it's super long, that's why I use already initialized DBs in the tests.
Using `InitSpatialMetadata(1)` could be interesting, though it would be a breaking change (or we could just let the use pass a parameter to `load_spatialite`).
Anyway, I agree it should be consistent between SpatiaLite and gpkg, so I will see what's the best option.

Reply:
In the end I added the possibility to pass an `init_mode` to the function, so it is possible to initialize the table with no SRID and then add the relevant ones manually. Unfortunately, it is not possible to pass an argument to the `sqlalchemy.event.listen` function, so if someone wants to use this argument he has to either create a new function that calls `load_spatialite` with an hard-coded `init_mode` or call `load_spatialite` manually after the connection is created.